<?xml version="1.0"?>
<launch>
    <arg name="roomba" />
    <arg name="robot_frame_id" default="$(arg roomba)/base_link" />
    <arg name="goal_dist" default="1.0" />
    <arg name="right_dist" default="0.2" />

    <arg name="map_file" default="$(find roomba_visualizer)/map/map.yaml" />

    <group ns="$(arg roomba)" >
        <node pkg="map_server" type="map_server" name="map_server" args="$(arg map_file)" />
        <include file="$(find multi_robots)/launch/launch_devices.launch" >
            <arg name="roomba" default="$(arg roomba)" />
        </include>
        <include file="$(find roomba_visualizer)/launch/amcl.launch" >
            <arg name="roomba" default="$(arg roomba)" />
        </include>

        <node pkg="dwa_planner" type="dwa_planner" name="dwa_planner">
            <rosparam command="load" file="$(find navigator)/config/dwa/dwa_param.yaml"/>
            <param name="ROBOT_FRAME" value="$(arg robot_frame_id)" />
            <remap from="/local_map" to="local_map"/>
            <remap from="/scan" to="corrected_scan"/>
            <remap from="/local_goal" to="local_goal"/>
            <remap from="/cmd_vel" to="local_path/cmd_vel"/>
            <remap from="/odom" to="roomba/corrected_odometry"/>
        </node>

        <node pkg="navigator" type="message_reviser_node" name="message_reviser" />
        <node pkg="navigator" type="waypoints_manager_node" name="waypoint_manager">
            <rosparam command="load" file="$(find navigator)/config/waypoints/dkan_all/$(arg roomba).yaml" />
            <param name="GOAL_THRESHOLD" value="$(arg goal_dist)" />
            <param name="RIGHT_DISTANCE" value="$(arg right_dist)" />
        </node>
    </group>
    <include file="$(find navigator)/launch/bag_record.launch">
        <arg name="roomba" value="$(arg roomba)" />
    </include>

</launch>
